File: app/src/main/java/com/elegion/android/template/ui/login/LoginActivity.java
package com.elegion.android.template.ui.login;

import android.content.Context;
import android.content.Intent;
import android.support.annotation.NonNull;
import android.support.v4.app.Fragment;

import com.elegion.android.template.ui.base.activity.SingleFragmentActivity;

/**
 * @author Mike
 */
public class LoginActivity extends SingleFragmentActivity {

    public static Intent makeIntent(@NonNull Context context) {
        return new Intent(context, LoginActivity.class);
    }

    @Override
    public Fragment createFragment() {
        return LoginFragment.newInstance();
    }
}

File: app/src/main/kotlin/com/elegion/android/template/ui/login/LoginActivity.kt
package com.elegion.android.template.ui.login

import android.content.Context
import android.content.Intent
import android.support.v4.app.Fragment

import com.elegion.android.template.ui.base.activity.SingleFragmentActivity

class LoginActivity : SingleFragmentActivity() {

    override fun createFragment(): Fragment = LoginFragment.newInstance()

    companion object {
        fun makeIntent(context: Context) = Intent(context, LoginActivity::class.java)
    }
}
